src/main/java/gov/nist/secauto/oscal/lib/profile/resolver/selection/Import.java

```diff
@@ -158,41 +158,48 @@ private static void generateMetadata(
         resolvedMetadata = new Metadata();
         resolvedCatalog.setMetadata(resolvedMetadata);
       }
+      resolveMetadata(importedMetadata, resolvedMetadata, indexer);
+    }
+  }
 
-      String importedVersion = importedMetadata.getOscalVersion();
-      if (importedVersion != null) {
-        Version importOscalVersion = VersionUtil.parseVersion(importedVersion, null, null);
+  private static void resolveMetadata(
+      @NonNull Metadata imported,
+      @NonNull Metadata resolved,
+      @NonNull IIndexer indexer) {
+    String importedVersion = imported.getOscalVersion();
+    if (importedVersion != null) {
+      Version importOscalVersion = VersionUtil.parseVersion(importedVersion, null, null);
 
-        Version resolvedCatalogVersion
-            = VersionUtil.parseVersion(resolvedMetadata.getOscalVersion(), null, null);
+      Version resolvedCatalogVersion
+          = VersionUtil.parseVersion(resolved.getOscalVersion(), null, null);
 
-        if (importOscalVersion.compareTo(resolvedCatalogVersion) > 0) {
-          resolvedMetadata.setOscalVersion(importOscalVersion.toString());
-        }
+      if (importOscalVersion.compareTo(resolvedCatalogVersion) > 0) {
+        resolved.setOscalVersion(importOscalVersion.toString());
```

Comment:
Minor, non-blocking: should we be concerned about that `oscal-version` is a [string datatype](https://pages.nist.gov/OSCAL/reference/latest/profile/xml-reference/#/profile/metadata/oscal-version) and that comparison on non-semverish strings could throw up stack traces here? Minor but need to investigate further about what happens with setting and comparing `apple` and `banana` (not that I would do that like the funny, zany developer I am but the schema allows it).

Reply:
Sure. I'd be happy to have better logic for this. Maybe create an issue to investigate?

Reply:
https://github.com/usnistgov/liboscal-java/issues/158